airbyte-workers/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java

```diff
@@ -163,6 +167,28 @@ public Process create(final String jobId,
     }
   }
 
+  private static String createContainerName(final String fullImagePath, final String jobId, final int attempt) {
+    final var versionDelimiter = ":";
+    final var noVersion = fullImagePath.split(versionDelimiter)[0];
+
+    final var dockerDelimiter = "/";
+    final var nameParts = noVersion.split(dockerDelimiter);
+    var imageName = nameParts[nameParts.length - 1];
+
+    final var randSuffix = RandomStringUtils.randomAlphabetic(5).toLowerCase();
+    final String suffix = "sync" + "-" + jobId + "-" + attempt + "-" + randSuffix;
+
+    var podName = imageName + "-" + suffix;
+    final var podNameLenLimit = 128;
+    if (podName.length() > podNameLenLimit) {
+      final var extra = podName.length() - podNameLenLimit;
+      imageName = imageName.substring(extra);
+      podName = imageName + "-" + suffix;
+    }
+
+    return podName;
```

Comment:
nit but should this be called `containerName` or should keep the same for future refactor?

Reply:
I'll keep it for a future refactor.

Reply:
which I'll do in the next few days.